sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -32,6 +32,10 @@ class SparkOptimizer(
   override def defaultBatches: Seq[Batch] = (preOptimizationBatches ++ super.defaultBatches :+
     Batch("Optimize Metadata Only Query", Once, OptimizeMetadataOnlyQuery(catalog)) :+
     Batch("Extract Python UDFs", Once,
+      PullOutPythonUDFInJoinCondition,
```

Comment:
Rename it to ExtractPythonUDFFromJoinCondition ?

Reply:
A separate question. Do we have a test case that ensure the local predicate that only contains PythonUDF will be pushed down Join? cc @cloud-fan

Reply:
not sure. maybe we can add one in `FilterPushdownSuite`